common4j/src/main/com/microsoft/identity/common/java/cache/AbstractAccountCredentialCache.java

```diff
@@ -140,6 +142,8 @@ protected List<Credential> getCredentialsFilteredByInternal(@Nullable final Stri
         final boolean mustMatchOnRealm = !StringUtil.isNullOrEmpty(realm);
         final boolean mustMatchOnTarget = !StringUtil.isNullOrEmpty(target);
         final boolean mustMatchOnClientId = !StringUtil.isNullOrEmpty(clientId);
+        final boolean mustMatchOnApplicationIdentifier = !StringUtil.isNullOrEmpty(applicationIdentifier);
+        final boolean mustMatchOnMamEnrollmentIdentifier = !StringUtil.isNullOrEmpty(mamEnrollmentIdentifier);
```

Comment:
Would this bypass the MAM ID check if the request does NOT have MAM ID, but the AT cache has?

Reply:
That's correct.  No enrollment id would be returned.